server/media/media_api.py

```diff
@@ -41,31 +41,56 @@ def get_instance_media(filename):
 # TODO: Restrict to logged-in scenario creators
 @api.post("/<path:filename>")
 def post_instance_media(filename):
-    """ Allows users to upload media files to the server. """
+    """ Allows users to upload media files to the server. Returns a MediaData-JSON. """
+    if not filename:
+        return "Invalid request: No filename provided", 400
+    elif filename == "txt":
+        return __handle_raw_text()
+    else:
+        return __handle_file_upload()
+
+
+def __handle_file_upload():
     if "file" not in request.files:
-        return "No file part in request", 400
+        return "Invalid request: Missing attribute 'file' in request", 400
 
     file = request.files["file"]
 
     if not file.filename:
-        return "No file provided", 400
+        return "Invalid request: No file provided", 400
 
     filename = secure_filename(file.filename)
     extension = os.path.splitext(filename)[1]
     reference_path = "media/instance/"
-
     if extension in MediaData.ALLOWED_IMAGE_EXTENSIONS:
         reference_path += "image"
+        result = MediaData.new_image(image_reference=f"{reference_path}/{filename}",
+                                     title=request.form.get("title", default=None))
     elif extension in MediaData.ALLOWED_VIDEO_EXTENSIONS:
         reference_path += "video"
+        result = MediaData.new_video(video_reference=f"{reference_path}/{filename}",
+                                     title=request.form.get("title", default=None))
     elif extension in MediaData.ALLOWED_AUDIO_EXTENSIONS:
         reference_path += "audio"
+        result = MediaData.new_audio(audio_reference=f"{reference_path}/{filename}",
+                                     title=request.form.get("title", default=None))
     elif extension in MediaData.ALLOWED_TEXT_EXTENSIONS:
         reference_path += "text"
+        result = MediaData.new_text_file(text_reference=f"{reference_path}/{filename}",
+                                         title=request.form.get("title", default=None))
     else:
-        return "Forbidden file format", 400
+        return "Forbidden file format", 403
 
     save_path = os.path.join(current_app.root_path, reference_path, filename)
     # TODO: Maybe also check if extension matches contents of file
```

Comment:
how do we handle this todo? another issue or don't relevant at this point and delete it